tony-core/src/main/java/com/linkedin/tony/TonyClient.java

```diff
@@ -386,6 +387,53 @@ public void initTonyConf(Configuration tonyConf, CommandLine cliParser) throws I
     tonyConf.addResource(new Path(tonyConfDir + File.separatorChar + Constants.TONY_SITE_CONF));
   }
 
+  /**
+   * This helper method parses and updates tonyConf's jobtype.resources fields
+   *  - If the file is remote, keep as is.
+   *  - If the file is local and is a file, upload to remote fs and replace this entry with address of uploaded file.
+   *  - If the file is local and is a directory, zip the directory, upload to remote fs and replace entry with the
+   *  address of the uploaded file.
+   **/
+  private void processTonyConfResources(Configuration tonyConf, FileSystem fs) throws IOException {
```

Comment:
Can we add some unit tests for this method?

Reply:
Yah, wanted to throw out the idea and make sure the API is fine before adding those tests.